web-common/src/features/dashboards/proto-state/toProto.ts

```diff
@@ -187,28 +181,78 @@ function toTimeGrainProto(timeGrain: V1TimeGrain) {
   }
 }
 
-function toFilterCondProto(conds: Array<MetricsViewFilterCond>) {
-  return conds.map(
-    (include) =>
-      new MetricsViewFilter_Cond({
-        name: include.name,
-        like: include.like,
-        in: include.in?.map(
-          (v) =>
-            (v === null
-              ? new Value({
-                  kind: {
-                    case: "nullValue",
-                    value: NullValue.NULL_VALUE,
-                  },
-                })
-              : new Value({
-                  kind: {
-                    case: "stringValue",
-                    value: v as string,
-                  },
-                })) ?? [],
-        ),
-      }),
-  );
+function toExpressionProto(expression: V1Expression | undefined) {
+  if (!expression) return undefined;
+  if ("ident" in expression) {
+    return new Expression({
+      expression: {
+        case: "ident",
+        value: expression.ident as string,
+      },
+    });
+  }
+  if ("val" in expression) {
+    return new Expression({
+      expression: {
+        case: "val",
+        value: toPbValue(expression.val),
+      },
+    });
+  }
+  if (expression.cond) {
+    return new Expression({
+      expression: {
+        case: "cond",
+        value: new Condition({
+          op: ToProtoOperationMap[
+            expression.cond.op ?? V1Operation.OPERATION_UNSPECIFIED
+          ],
+          exprs: expression.cond.exprs?.map((e) => toExpressionProto(e)) ?? [],
+        }),
+      },
+    });
+  }
+  return undefined;
+}
+
+function toPbValue(val: unknown) {
+  if (val === null) {
+    return new Value({
+      kind: {
+        case: "nullValue",
+        value: NullValue.NULL_VALUE,
+      },
+    });
+  }
+  switch (typeof val) {
+    case "string":
+      return new Value({
+        kind: {
+          case: "stringValue",
+          value: val,
+        },
+      });
+    case "number":
+      return new Value({
+        kind: {
+          case: "numberValue",
+          value: val,
+        },
+      });
+    case "boolean":
+      return new Value({
+        kind: {
+          case: "boolValue",
+          value: val,
+        },
+      });
+    // TODO: other options are not currently in a filter. but we might need them in future
+  }
+  // force as string. this is the older behaviour
```

Comment:
Sounds good. Updated to force convert using `JSON.stringify`. We should do a global refactor of dimension value to be anything and not just string. This can hopefully be fixed then.